src/eccs/Q3ShorEcc.cpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * This file is part of JKQ QFR library which is released under the MIT license.
+ * See file README.md or go to http://iic.jku.at/eda/research/quantum/ for more information.
+ */
+
+#include "eccs/Q3ShorEcc.hpp"
+
+#include <chrono>
+//#include <stdlib.h>
+
+//3 data qubits, 2 for measuring -> 5 qubits per physical qubit
+Q3ShorEcc::Q3ShorEcc(qc::QuantumComputation& qc): Ecc({EccID::Q3Shor, 5, 2, Q3ShorEcc::getEccName()}, qc) {}
+
+void Q3ShorEcc::writeEccEncoding() {
+	const int nQubits = qc.getNqubits();
+
+    for(int i=0;i<nQubits;i++) {
+        writeCnot(i, i+nQubits);
+        writeCnot(i, i+2*nQubits);
+    }
+}
+
+void Q3ShorEcc::measureAndCorrect() {
+    const int nQubits = qc.getNqubits();
+    for(int i=0;i<nQubits;i++) {
+
+        qcMapped.h(i+3*nQubits);
+        qcMapped.h(i+4*nQubits);
+        dd::Control c3 = createControl(i+3*nQubits, true);
+        dd::Control c4 = createControl(i+4*nQubits, true);
```

Comment:
```suggestion
        auto c3 = createControl(i+3*nQubits, true);
        auto c4 = createControl(i+4*nQubits, true);
```
You can use `auto` in cases like these.